### PR TITLE
boilerplate: simplify prow config

### PR DESCRIPTION
**Pull request by 2uasimojo**
- Get `build_root` from the repo (prereq: https://github.com/openshift/boilerplate/pull/76)
- Since https://github.com/openshift/boilerplate/pull/72 / https://github.com/openshift/boilerplate/pull/73 and https://github.com/openshift/release/pull/12778, we're set up for boilerplate consumers to use an ImageStream from quay, so we no longer need to publish the build image from here. Nix it.
- Remove unnecessary ocp 4.4 base image.

### CODE REVIEW

File: ci-operator/config/openshift/boilerplate/openshift-boilerplate-master.yaml [hunk 1]
```diff
@@ -1,24 +1,10 @@
...
 build_root:
-  project_image:
-    context_dir: config/
-    dockerfile_path: Dockerfile
+  from_repository: true
 images:
 - context_dir: config/
   dockerfile_path: Dockerfile
-  from: openshift_release_golang-1.13
+  from: src
```
Comment (arjunrn) [inline ci-operator/config/openshift/boilerplate/openshift-boilerplate-master.yaml:6]: I assume this means use the `.ci-operator.yaml` to get the base image.

**Comment by 2uasimojo** [comment]
/retest

prereq landed

File: ci-operator/config/openshift/boilerplate/openshift-boilerplate-master.yaml (hunk 1 above)
Comment (2uasimojo) [reply ci-operator/config/openshift/boilerplate/openshift-boilerplate-master.yaml:6]: Yeah, `src` is a special name given to the image produced by `build_root`. Or is it `root`? I think (per the [docs](https://docs.ci.openshift.org/docs/architecture/ci-operator/#building-artifacts)) that either will work here, but let me check with dptp...

**Comment by 2uasimojo** [comment]
/retest

cmon now

File: ci-operator/config/openshift/boilerplate/openshift-boilerplate-master.yaml (hunk 1 above)
Comment (petr-muller) [reply ci-operator/config/openshift/boilerplate/openshift-boilerplate-master.yaml:6]: `src` is `root` with the repository cloned into it.

Comment (2uasimojo) [reply ci-operator/config/openshift/boilerplate/openshift-boilerplate-master.yaml:6]: Cool, that's what we want here ✓

Thanks @petr-muller.

**Comment by 2uasimojo** [comment]
Current failures are being addressed by https://github.com/openshift/boilerplate/pull/77

**Comment by 2uasimojo** [comment]
/retest

https://github.com/openshift/boilerplate/pull/77 landed.

**Pull request merged**